sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -74,10 +74,20 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   }
 
   /**
-   * @return All metrics containing metrics of this SparkPlan.
+   * @return Executor side metrics of this SparkPlan.
    */
   def metrics: Map[String, SQLMetric] = Map.empty
 
+  /**
+   * @return Driver side metrics of this SparkPlan which no need to send them to executor.
```

Comment:
please add more comment about the difference between `metrics` and `driverMetrics`. e.g. one needs to be serialized and sent to executors and one not.

Reply:
Copy that, done in 01c0179.